src/classes/ADDR_Addresses_UTIL.cls

```diff
@@ -46,255 +46,298 @@ public with sharing class ADDR_Addresses_UTIL {
     public static Boolean isAccAddressEmpty(Account acc1) {
-        if (acc1 == null) 
-           return true;
+        if (acc1 == Null) {
+            return True;
+        }
         
-        Boolean isEmpty = 
-           acc1.BillingStreet == null &&
-           acc1.BillingCity == null &&
-           acc1.BillingState == null &&
-           acc1.BillingPostalCode == null &&
-            // note that we decided to ignore country so that a default value won't create unnecessary address objects
-           (acc1.BillingCountry == null || ADDR_Addresses_UTIL.isStateCountryPicklistsEnabled) &&
-           // only test StateCode if picklists enabled.
-           (!ADDR_Addresses_UTIL.isStateCountryPicklistsEnabled || acc1.get('BillingStateCode') == null);
+        Boolean isEmpty = String.isBlank(acc1.BillingStreet) &&
+                            String.isBlank(acc1.BillingCity) &&
+                            String.isBlank(acc1.Billing_County__c) &&
+                            String.isBlank(acc1.BillingState) &&
+                            String.isBlank(acc1.BillingPostalCode) &&
+                            //Note that we decided to ignore country so that a default value won't create unnecessary address objects. 
+                            (String.isBlank(acc1.BillingCountry) || ADDR_Addresses_UTIL.isStateCountryPicklistsEnabled) &&
+                            //Only test StateCode if picklists enabled.
+                            (!ADDR_Addresses_UTIL.isStateCountryPicklistsEnabled || acc1.get('BillingStateCode') == Null);
         return isEmpty;   
     }
     
     /*******************************************************************************************************
-    * @description Returns whether two strings are equal, using a case sensitve comparison
-    * @param str1 The first string
-    * @param str2 The second string
+    * @description Returns whether two strings are equal, using a case sensitve comparison.
+    * @param str1 The first string.
+    * @param str2 The second string.
     * @return Boolean 
     ********************************************************************************************************/
-    public static Boolean equalsCaseSensitive(string str1, string str2) {
-        if (str1 == null)
-            return str2 == null;
-        if (str2 == null)
-            return false;
+    public static Boolean equalsCaseSensitive(String str1, String str2) {
+        if (str1 == Null) {
+            return str2 == Null;
+        }
+            
+        if (str2 == Null) {
+            return False;
+        }
+        
         return str1.equals(str2);
     }
     
     /*******************************************************************************************************
-    * @description Compare two address objects address fields and support fuzzy matching; our definition of 
+    * @description Compare two Address records' address fields and support fuzzy matching. Definition of 
```

Comment:
@MatthewBlanski  thank you so much! Yes I really disliked the use of the word object because it didn't really mean that we created new objects. It made me a bit confusing.